### DISCUSSION TITLE
An API to support query of call logs

### DISCUSSION THREAD

**Discussion by mianaliasjad** [vestrel00/contacts-android#265] [Ideas]
To query the call logs we use `CallLog.Calls.CONTENT_URI` maybe we can add this functionality in this library. But i think it should be added as a seprate module maybe? 

Something like that:

`val callLogs = CallLog(context).query().find()`

**Comment by vestrel00** [comment]
Hey @mianaliasjad, thanks for starting this discussion!

I spent some time reading through the documentation around that Android API; https://developer.android.com/reference/android/provider/CallLog.Calls

What I can say is that it is definitely a feature that could be a part of this library and it would not be out of place. In terms of scope, it would be similar to the BlockedNumbers API that this library already covers. However, unlike blocked numbers where its purpose is closely tied to contacts applications and phone/dialer applications, call logs seem to be leaning a bit more heavily towards just phone/dialer apps. This is supported by the fact that the AOSP Contacts app, Google Contacts app, and Samsung Contacts app all provide an entry point to blocked numbers but not call logs.

Given that this library’s focus is mainly on providing APIs to allow anyone to recreate a contacts application, my opinion on adding a call log wrapper api is that it is not a high priority for this library to implement (for now) as there are other more important things to do in the [project roadmap](https://github.com/vestrel00/contacts-android/wiki/Project-Roadmap) to get to the first true production-ready 1.0.0 version of this library. 

I do think that this is a feature that will ultimately end up in this library, but perhaps not before the other already-planned stuff. Let’s keep this discussion open so that others can participate if they are interested. If I get more people requesting this feature, then I may bump up priority and implement it sooner than planned 😁
